src/libraries/Common/tests/System/Security/Cryptography/X509Certificates/X509CertificateLoaderPkcs12CollectionTests.cs

```diff
@@ -786,6 +786,28 @@ public void LoadWithDuplicateAttributes(bool allowDuplicates)
             }
         }
 
+        [Theory]
+        [InlineData(false, false)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        [InlineData(true, true)]
+        public void LoadWithLegacyProvider(bool preserveStorageProvider, bool ephemeralIfPossible)
+        {
+            Pkcs12LoaderLimits limits = new Pkcs12LoaderLimits { PreserveStorageProvider = preserveStorageProvider };
+            X509KeyStorageFlags flags = ephemeralIfPossible ? EphemeralIfPossible : X509KeyStorageFlags.DefaultKeySet;
+            bool expectLegacy = ((int)flags & 0x20) == 0 && preserveStorageProvider;
```

Comment:
I don't quite grok the intention of this. Why is this not just `flags & X509KeyStorageFlags.EphemeralKeySet`?

Reply:
Oh. Because .NET Framework doesn't have that enum value until 4.8.

Pull it up in to a named const?

Reply:
Because that file is included in TFMs that don't have that flag :smile: